extensions/windows-event-log/wel/WindowsEventLog.cpp

```diff
@@ -129,34 +133,30 @@ std::string WindowsEventLogMetadataImpl::getEventData(EVT_FORMAT_MESSAGE_FLAGS f
     return event_data;
   }
 
-  if (!EvtFormatMessage(metadata_ptr_, event_ptr_, 0, 0, NULL, flags, string_buffer_size, string_buffer, &string_buffer_used)) {
+
+  if (!EvtFormatMessage(metadata_ptr_, event_ptr_, 0, 0, NULL, flags, string_buffer_size, string_buffer.get(), &string_buffer_used)) {
     result = GetLastError();
     if (ERROR_INSUFFICIENT_BUFFER == result) {
       string_buffer_size = string_buffer_used;
 
-      string_buffer = (LPWSTR) malloc(string_buffer_size * sizeof(WCHAR));
-
-      if (string_buffer) {
+      string_buffer.reset((LPWSTR) malloc(string_buffer_size * sizeof(WCHAR)));
 
-        if ((EvtFormatMessageKeyword == flags))
-          string_buffer[string_buffer_size - 1] = L'\0';
-
-        EvtFormatMessage(metadata_ptr_, event_ptr_, 0, 0, NULL, flags, string_buffer_size, string_buffer, &string_buffer_used);
-        if ((EvtFormatMessageKeyword == flags))
-          string_buffer[string_buffer_used - 1] = L'\0';
-        std::wstring str(string_buffer);
-        event_data = std::string(str.begin(), str.end());
-        free(string_buffer);
-      }
+      EvtFormatMessage(metadata_ptr_, event_ptr_, 0, 0, NULL, flags, string_buffer_size, string_buffer.get(), &string_buffer_used);
     }
   }
+  if ((EvtFormatMessageKeyword == flags))
```

Comment:
minor, and old code, but why do we have double `(())` here?

Reply:
done